wandb/sdk/wandb_run.py

```diff
@@ -1436,16 +1496,15 @@ def _on_init(self) -> None:
 
     def _on_start(self) -> None:
         # TODO: make offline mode in jupyter use HTML
-        if self._settings._offline:
-            wandb.termlog("Offline run mode, not syncing to the cloud.")
-
         if self._settings._offline:
             wandb.termlog(
                 (
                     "W&B syncing is set to `offline` in this directory.  "
-                    "Run `wandb online` to enable cloud syncing."
+                    "Run `wandb online` or set WANDB_MODE=online to enable cloud syncing."
                 )
             )
+        if self._settings.code_dir is not None:
+            self.log_code(self._settings.code_dir)
```

Comment:
@raubitsj this gives us `wandb.Settings(code_dir="xxx")` and `WANDB_CODE_DIR=XXX` triggering automatic calling of `log_code`.  I decided not to implement `.gitignore` parsing for now as it adds a bunch of complexity.   In the future I think the logic should be:

1. Only include `.py` files and exclude any files in `os.sep + wandb + os.sep/*`.
2. Look for a `.gitignore` in the root and every subdirectory as you crawl the file system.
3. Adhere to the `.gitignore` from the directory it was found... i.e. include all files that don't match the `.gitignore`
4. To do this right, we should write our own `.gitignore` to the `wandb` directory...